Tests/MapboxMapsTests/MapView/Integration Tests/AnnotationManagerIntegrationTests.swift

```diff
@@ -61,10 +61,10 @@ internal class AnnotationManagerIntegrationTestCase: MapViewIntegrationTestCase
             // Given
             let annotation = PointAnnotation(coordinate: mapView.cameraState.center)
             let requiredIndex = 3
-            let position = LayerPosition(above: nil, below: nil, at: requiredIndex)
             let annotationManager = AnnotationManager(for: mapView,
                                                       mapEventsObservable: mapView.mapboxMap,
-                                                      with: self)
+                                                      with: self,
+                                                      options: AnnotationOptions(layerPosition: .at(requiredIndex)))
```

Comment:
I feel like we need another test that specifically adds an annotation above location indicator Layer or vice-versa. And then we can get the stack and check for the order there as the assertion. That was a specific use case and would be an impactful test.

Further, this PR is going to warrant a Migration Guide change. Can we make that and link that PR as part of this ticket somewhere.

Reply:
Can we address this with upcoming annotation changes? /cc @nishant-karajgikar?